tiledb/sm/group/group.cc

```diff
@@ -83,6 +84,48 @@ Group::Group(
   memory_tracker_->set_type(MemoryTrackerType::GROUP);
 }
 
+Status Group::create(ContextResources& resources, const URI& uri) {
+  // Create group URI
+  if (uri.is_invalid())
+    throw GroupStatusException(
+        "Cannot create group '" + uri.to_string() + "'; Invalid group URI");
+
+  // Check if group exists
+  bool exists;
+  throw_if_not_ok(is_group(resources, uri, &exists));
+  if (exists) {
+    throw GroupStatusException(
+        "Cannot create group; Group '" + uri.to_string() + "' already exists");
+  }
+
+  std::mutex object_create_mtx;
+  std::lock_guard<std::mutex> lock{object_create_mtx};
```

Comment:
When a mutex is defined as an ordinary local variable, it doesn't do anything useful.

The old member variable `object_create_mtx_` from which this was taken appears both in `group_create` but also in `array_create`. In order to preserve this behavior, references to the mutex will have to appear in two separate source directories. This is a problem.

It's not at all clear to me that we even need or want this lock. What to do about any changes to behavior are out of scope. To preserve behavior, though, we'll need a mutex somewhere.
* It doesn't go properly in either `sm\array` or `sm\group`, but it could go in `sm\object`.
* It's a very narrow usage that we don't want to generalize. So the mutex should be declared in a header all by itself. The explanation for why it's all by its lonesome should be verbosely documented in the header.
* The definition for the mutex (in some `.cc` file) doesn't need to be in a separate file.
* The header should be included in exactly three locations: one for its definition, one for array, and one for group. The one for array will need to be in `storage_manager.cc` for the moment until `array_create` is moved.